core/lib/soci_sqlite3/session.cpp

```diff
@@ -125,17 +180,28 @@ sqlite3_session_backend::sqlite3_session_backend(
     post_connection(dbname, connection_flags, timeout, synchronous);
 
     //Set password
-    if (!passKey.empty())
+    if (!passKey.empty() || !newPassKey.empty())
     {
-        res = sqlite3_key_v2(conn_, dbname.c_str(), passKey.c_str(), strlen(passKey.c_str()));
-        check_sqlite_err(conn_, res, "Failed to encrypt database. ");
+        if (!passKey.empty()){
```

Comment:
I need to read all those tests and have a look at the generated conditional graph. That seems like a lot of redundant testing.

Reply:
Yes please do !
Again here it's more about easily and clearly see the flow, let's make it green then let's make it gold ! :trollface: